src/mono/wasm/runtime/startup.ts

```diff
@@ -842,48 +880,62 @@ let configLoaded = false;
         return;
     }
     if (runtimeHelpers.diagnostic_tracing) console.debug("MONO_WASM: mono_wasm_load_config");
     try {
         const resolveSrc = runtimeHelpers.locateFile(configFilePath);
         const configResponse = await runtimeHelpers.fetch_like(resolveSrc);
         const configData: MonoConfig = (await configResponse.json()) || {};
+        if (configData.environment_variables && typeof (configData.environment_variables) !== "object")
+            throw new Error("Expected config.environment_variables to be unset or a dictionary-style object");
+
         // merge
         configData.assets = [...(config.assets || []), ...(configData.assets || [])];
+        configData.environment_variables = { ...(config.environment_variables || {}), ...(configData.environment_variables || {}) };
         config = runtimeHelpers.config = Module.config = Object.assign(Module.config as any, configData);
-
-        // normalize
-        config.environment_variables = config.environment_variables || {};
-        config.assets = config.assets || [];
-        config.runtime_options = config.runtime_options || [];
-        config.globalization_mode = config.globalization_mode || "auto";
-        if (config.enable_debugging)
-            config.debug_level = config.enable_debugging;
-
-        if (typeof (config.environment_variables) !== "object")
-            throw new Error("Expected config.environment_variables to be unset or a dictionary-style object");
+        normalize();
 
         if (Module.onConfigLoaded) {
             try {
                 await Module.onConfigLoaded(<MonoConfig>runtimeHelpers.config);
+                normalize();
             }
             catch (err: any) {
                 _print_error("MONO_WASM: onConfigLoaded() failed", err);
                 throw err;
             }
         }
-        runtimeHelpers.diagnostic_tracing = !!runtimeHelpers.config.diagnostic_tracing;
-        runtimeHelpers.enable_debugging = runtimeHelpers.config.enable_debugging ? runtimeHelpers.config.enable_debugging : 0;
-        configLoaded = true;
+        afterConfigLoaded.promise_control.resolve();
     } catch (err) {
         const errMessage = `Failed to load config file ${configFilePath} ${err}`;
         abort_startup(errMessage, true);
         config = runtimeHelpers.config = Module.config = <any>{ message: errMessage, error: err, isError: true };
         throw err;
     }
+
+    function normalize() {
+        // normalize
+        config.environment_variables = config.environment_variables || {};
+        config.assets = config.assets || [];
+        config.runtime_options = config.runtime_options || [];
+        config.globalization_mode = config.globalization_mode || "auto";
+        if (config.enable_debugging)
+            config.debug_level = config.enable_debugging;
+        if (config.debug_level === undefined && BuildConfiguration === "Debug") {
+            config.debug_level = -1;
+        }
```

Comment:
Isn't this a change in behavior? We are disabling this by default for Debug config?
Also, this seems disconnected from the debug_level stuff in `mono_wasm_load_runtime`.

Reply:
```	 * debug_level > 0 enables debugging and sets the debug log level to debug_level
	 * debug_level == 0 disables debugging and enables interpreter optimizations
	 * debug_level < 0 enabled debugging and disables debug logging.
```
so we are enabling it for debug build configuration. Yes that's change in default behavior.

In `mono_wasm_load_runtime` we are just propagating value from `config` if it was not provided as argument.

Reply:
Now I did the same also for `diagnosticTracing` on debug build of runtime.